### PR TITLE
ACS-8151 Answer API returnig HX Insight response

### PR DESCRIPTION
**Pull request by mpichura**

### CODE REVIEW

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClientTest.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ void shouldThrowOnNotExpectedStatusCode()
         given(httpClient.send(any(), any())).willReturn(response);
 
         // when
-        assertThrows(EndpointClientErrorException.class, () -> hxInsightClient.askQuestion(
+        assertThrows(WebScriptException.class, () -> hxInsightClient.askQuestion(
```
Comment (mpichura) [inline hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClientTest.java:118]: @Pawel-608 I suspect this change should not be here?

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClientTest.java [hunk 2]
```diff
@@ -154,7 +154,7 @@ void shouldThrowOnNotExpectedStatusCodeWhenGettingAnswer()
         given(config.getAnswerUrl()).willReturn("http://hxinsight/question/%s/answer");
 
         // when + then
-        assertThrows(EndpointClientErrorException.class, () -> hxInsightClient.getAnswer(questionId));
+        assertThrows(WebScriptException.class, () -> hxInsightClient.getAnswer(questionId));
```
Comment (mpichura) [inline hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClientTest.java:157]: @Pawel-608 I suspect this change should not be here?

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelationTest.java [hunk 3]
```diff
@@ -51,11 +63,31 @@ void shouldCallHxInsightClientGetAnswer()
...
-        objectUnderTest.readAll(questionId, null);
+        CollectionWithPagingInfo<AnswerModel> answerResponse = objectUnderTest.readAll(questionId, mockParameters);
 
         // then
         then(mockHxInsightClient).should().getAnswer(questionId);
+        Collection<AnswerModel> answerResponseEntries = answerResponse.getCollection();
+        assertEquals(1, answerResponseEntries.size());
+        AnswerModel answer = answerResponseEntries.iterator().next();
+        assertEquals(hXAnswer.getAnswer(), answer.getAnswer());
+        assertEquals(hXAnswer.getQuestionId(), answer.getQuestionId());
```
Comment (tpage-alfresco) [inline hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelationTest.java:80]: If you construct an expected object then I think you can do a single assertion here:
```
assertEquals(expected, answerResponseEntries);
```

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/model/AnswerModelTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        String answerText = "answer";
+        answer.setAnswer(answerText);
+        String questionId = "questionId";
+        answer.setQuestionId(questionId);
+        AnswerResponse.Reference reference = new AnswerResponse.Reference();
+        String referenceId = "referenceId";
+        reference.setReferenceId(referenceId);
+        String referenceText = "referenceText";
+        reference.setTextReference(referenceText);
+        answer.setReferences(Set.of(reference));
```
Comment (tpage-alfresco) [inline hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/model/AnswerModelTest.java:52]: Maybe consider adding an extra constructor or builder to `AnswerResponse` for unit testing.

**Review by tpage-alfresco — APPROVED** [review]
All looked good. A few suggestions but nothing essential.

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/model/AnswerModelTest.java (hunk 4 above)
Comment (mpichura) [reply hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/model/AnswerModelTest.java]: Good point - I added a builder and changed that here and in other test as well.

File: hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelationTest.java (hunk 3 above)
Comment (mpichura) [reply hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelationTest.java]: Good point - changed.

**Pull request merged**